### ISSUE TITLE
Zoom direction: UI mismatch, start locations and directory defaults

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#7964]
* KOReader version: 2021-06-24
* Device: Kindle PW3 (FW 5.9.7)

#### Issue
Here is the UI for reference. Below I use a simple convention for `Direction`, e.g. `Direction` number 1 is the first choice from the left in the UI.
![Reader_2021-07-11_213336](https://user-images.githubusercontent.com/6279855/125217994-f2677800-e28f-11eb-91d9-f5c09316bdee.png)

1) In the UI, icons for `Direction` numbers 3 and 6 are swapped. I checked that manually - they work as mirrors to each other.
**Suggestion**: for simplicity, let's just swap the icon images in the UI only. However, I am not sure whether the numbers behind them are important in the code.

2) Sometimes, while in zoom with zoom factor other than 1.0 (maybe this happens with 1.0 too, but its impossible to notice), changing `Direction` does not bring you to corresponding starting corner. Sometimes, it jumps to a random corner. It is inconsistent, but I'll try to explain on a reproducible example. Here is an example file which I open on my Kindle which has 300 DPI: [minamoto-0112-0003-kcc.zip](https://github.com/koreader/koreader/files/6797868/minamoto-0112-0003-kcc.zip)
- Open a muPDF file. My defaults for `Zoom to` is to content and for `Direction` is number 1.
- Set `Zoom to` to manual (`%`)
- Change `Zoom factor` from 1.0 to 1.5. It will zoom to the top left corner. The page is divided into 4 corners (2 rows and 2 columns of corners to go through on a page).
- Set `Direction` to number 5. It will jump to (!) top right corner (instead of bottom right). However, the actual direction changed correctly (so if I were to tap/swipe forward it would continue to bottom-left, then to top-left).
- Tap number 1 again, it will jump to (!) bottom left corner. Tapping back and forth between number 1 and 5 will bring them again to these respective corners.

And there are many such "pairs" of weird jumps, which I can't understand how they work.

There are also pairs of `Direction` switches that I found to switch to proper starting locations (but I didn't test all pairs):
- All symmetrically placed numbers: 1 and 8, 2 and 7, 3 and 6 (in their mirrored behavior), 4 and 5. I noticed that these pairs have their starting locations on the same row, e.g. numbers 1 and 8 have it on the top row of corners.
- Pairs that have the same starting position (1 and 2, 7 and 8, etc)

Tapping the same `Direction` twice will actually bring you the the proper starting location on the page.
**Suggestion**: maybe we should force to jump to proper location when _any_ `Direction` is selected?

3) I guess, this issue is more related to `Zoom factor` (#7962), but its context is also related to `Direction` somewhat.
I apply the directory default:
```
["kopt_zoom_direction"] = 0,
```
where this number ranges from 7 to 0, which corresponds to `Direction` numbers 1 to 8, respectively.

When I open a new document from that directory:
- The UI shows respectively defined default underlined (even when `Zoom to` is set to content). E.g. for `["kopt_zoom_direction"] = 0`, number 8 is underlined. If I close the document and check metadata.lua, `["kopt_zoom_direction"] = 0`, so directory defaults did the work. I didn't find any other similar property (in case of UI and setting mismatch as happened before).
-  Manually selecting `Zoom to` to `%` and then `Zoom factor` to 1.5 (or whatever), works fine: the starting location is the top right corner as expected.
- However, if in a new document I do pinch to zoom (which I set to change `Zoom mode -> manual` and `Change zoom factor`), it zooms to 1.5 (which is my long-tap default), but the starting location will be top-left corner, as if for `Direction` number 1. This is true for any other `["kopt_zoom_direction"]` which all zoom to top left corner and their `Direction` is the same as that of number 1.

**Comment by poire-z** [comment]
Pinging @jperon

**Comment by NiLuJe** [comment]
Yep, the icons are indeed misnamed ;).

**Comment by poire-z** [comment]
Has this been fixed?

**Comment by mergen3107** [comment]
I can see that the icons where swapped. The rest is the same (unless there were changes that didn’t mention this ticket)

**Comment by NiLuJe** [comment]
Nope, nothing besides the icons & configurable being accurate has changed.
